pandas/core/generic.py

```diff
@@ -8327,24 +8327,95 @@ def _doc_parms(cls):
 """
 
 _cnum_doc = """
+Return %(desc)s over a DataFrame or Series axis.
+
+Returns a DataFrame or Series of the same size containing the %(desc)s.
 
 Parameters
 ----------
 axis : %(axis_descr)s
 skipna : boolean, default True
     Exclude NA/null values. If an entire row/column is NA, the result
-    will be NA
+    will be NA.
+*args : any, default None
+**kwargs : any, default None
+    Additional keywords have no effect but might be accepted for
+    compatibility with NumPy.
 
 Returns
 -------
 %(outname)s : %(name1)s\n
-
-
+%(examples)s
 See also
 --------
 pandas.core.window.Expanding.%(accum_func_name)s : Similar functionality
     but ignores ``NaN`` values.
+pandas.Series.%(outname)s : Return %(desc)s over Series axis.
+pandas.DataFrame.cummax : Return cumulative maximum over DataFrame axis.
+pandas.DataFrame.cummin : Return cumulative minimum over DataFrame axis.
+pandas.DataFrame.cumsum : Return cumulative sum over DataFrame axis.
+pandas.DataFrame.cumprod : Return cumulative product over DataFrame axis.
+"""
 
+_cummax_examples = """\
+Examples
+--------
+**DataFrame**
+
+Create a DataFrame:
+
+>>> df = pd.DataFrame([[9, 7, 9, 7],
+...                    [7, 5, 2, 7],
+...                    [3, 5, 2, 2],
+...                    [8, 0, 9, 0]],
+...                    columns=list('ABCD'))
+>>> df
+   A  B  C  D
+0  9  7  9  7
+1  7  5  2  7
+2  3  5  2  2
+3  8  0  9  0
+
+axis=None : Iterates over rows and finds the maximum value in each column.
+If value is larger than the previous maximum, updates it:
+
+>>> df.cummax(axis=None)
+   A  B  C  D
+0  9  7  9  7
+1  9  7  9  7
+2  9  7  9  7
+3  9  7  9  7
```

Comment:
If this docstring is reused, and we want to keep it this way, I think we should add examples for all methods.